airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -131,6 +135,43 @@ def poke(self, context: Context):
         else:
             return all(self._check_key(key) for key in self.bucket_key)
 
+    def execute(self, context: Context) -> None:
+        """
+        Defers to Trigger class to poll for state of the job run until
+        it reaches a failure state or success state
+        """
+        if not self.deferrable:
+            super().execute(context)
+        else:
+            if not self.poke(context=context):
+                self.defer(
+                    timeout=timedelta(seconds=self.timeout),
+                    trigger=S3KeyTrigger(
+                        bucket_name=cast(str, self.bucket_name),
+                        bucket_key=self.bucket_key,
+                        wildcard_match=self.wildcard_match,
+                        check_fn=self.check_fn,
```

Comment:
I'm curious about how this works. How does a function handle being serialized? Are you able to reconstruct the function after it has been stored as a string in the database?

Reply:
Thanks @syedahsn I have changed the logic for `check_fn` [here](https://github.com/apache/airflow/pull/31018/commits/d36487d24c5bbb9ab43e8b503ea1842853332be7) Please check